src/modules/IsnsServer.rb

```diff
@@ -351,13 +351,14 @@ def isnsadm(params, ret_result = false)
 
     def isnsadm_control
       if !@isctrlnode
-        @isctrlnode = isnsadm("--register control")
-        if !@isctrlnode
+        res = isnsadm("--register control", ret_result = true)
+
+        if res["exit"] != 0
           y2error("Registering as control node failed: #{res["stderr"]}; #{res["stdout"]}")
         end
       end
 
-      @isctrlnode
+      @isctrlnode = res["exit"] == 0
```

Comment:
I think it would fail if `@isctrlnode` is `true` (because `res` is not initialized).